x/perpetual/keeper/updated_leverage.go

```diff
@@ -16,6 +16,10 @@ func (k Keeper) UpdatedLeverage(ctx sdk.Context, mtp types.MTP) (sdk.Dec, error)
 	}
 	baseCurrency := entry.Denom
 	collateral_in_usdc := math.LegacyDec(mtp.Collateral)
+	if mtp.CollateralAsset != baseCurrency {
+		price := k.amm.EstimatePrice(ctx, mtp.CustodyAsset, baseCurrency)
```

Comment:
@cryptokage1996 price is queried twice line 20 and 25 can we remove this redundancy? thanks

Reply:
@cryptokage1996 can you also fix the conflicts in this PR thanks